WooCommerce/Classes/POS/ViewHelpers/CollectCashViewHelper.swift

```diff
@@ -54,25 +54,17 @@ final class CollectCashViewHelper {
         }
     }
 
-    func validateAmountOnSubmit(orderTotal: String,
+    func isPaymentButtonEnabled(orderTotal: String,
                                 textFieldAmountInput: String,
-                                onError: (String) -> Void) -> Bool {
-        let userInput = textFieldAmountInput.isNotEmpty ? textFieldAmountInput : "0"
-
+                                isLoading: Bool) -> Bool {
         guard let orderDecimal = parseCurrency(orderTotal),
-              let inputDecimal = parseCurrency(userInput) else {
-            onError(Localization.failedToCollectCashPayment)
-            return false
-        }
-
-        if inputDecimal < orderDecimal {
-            onError(Localization.cashPaymentAmountNotEnough)
+              let inputDecimal = parseCurrency(textFieldAmountInput.isNotEmpty ? textFieldAmountInput : "0") else {
             return false
         }
-        return true
+        return inputDecimal >= orderDecimal && !isLoading
```

Comment:
nit: I don't find this function specially readable, perhaps just by switching the order a bit and extracting logic from the guard makes it clearer? Could be entirely personal so feel free to ignore (untested!):

```
    func isPaymentButtonEnabled(orderTotal: String,
                                textFieldAmountInput: String,
                                isLoading: Bool) -> Bool {
        if isLoading { return false }

        let sanitizedInput = textFieldAmountInput.isNotEmpty ? textFieldAmountInput : "0"

        guard let orderDecimal = parseCurrency(orderTotal),
              let inputDecimal = parseCurrency(sanitizedInput) else {
            return false
        }

        return inputDecimal >= orderDecimal
    }
```